server/src/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -800,6 +804,7 @@ public Discoverer getMatchingDiscover(final Hypervisor.HypervisorType hypervisor
 
                 }
                 s_logger.info("server resources successfully discovered by " + discoverer.getName());
+                agentMSLB.propagateManagementServerListToAgents();
```

Comment:
@nvazquez this is not desirable, this will cause the list to be sent to all the hosts whenever any single agent connects.